test/integration/protocol_integration_test.cc

```diff
@@ -406,6 +406,37 @@ name: add-trailers-filter
   }
 }
 
+TEST_P(ProtocolIntegrationTest, AccessLogTest) {
+  if (upstreamProtocol() == Http::CodecType::HTTP3) {
+    return;
+  }
+
+  config_helper_.addConfigModifier(
+      [&](envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager&
+              hcm) -> void {
+        hcm.mutable_access_log_options()->set_flush_access_log_on_new_request(true);
+      });
+
+  useAccessLog("%RESPONSE_CODE% %ACCESS_LOG_TYPE%");
+  initialize();
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+  auto response = codec_client_->makeHeaderOnlyRequest(Http::TestRequestHeaderMapImpl{
+      {":method", "GET"}, {":path", "/test"}, {":scheme", "http"}, {":authority", "host.com"}});
+  waitForNextUpstreamRequest();
+  EXPECT_EQ(absl::StrCat("0 ", AccessLogType_Name(AccessLog::AccessLogType::DownstreamStart)),
```

Comment:
I think you can set the `access_log_flush_interval` to .5 seconds and wait for log entry to appear before sending the response on line 429. When you check for the DownstreamEnd log at the end, you may allow for additional periodic entries, in case the test is very slow in sending the response and another periodic log was emitted during this time.

Reply:
I'll try to implement this tomorrow, I hope this kind of test won't be flaky or something, not sure how long the test would really take in reality. Are there any other required changes, so I'll batch them in a single iteration? @adisuissa can you please review?

Reply:
I'll try to implement this tomorrow, I hope this kind of test won't be flaky or something, not sure how long the test would really take in reality. Are there any other required changes, so I'll batch them in a single iteration? @adisuissa can you please review?